src/m365/spo/commands/page/page-control-set.spec.ts

```diff
@@ -0,0 +1,325 @@
+        return Promise.resolve({});
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    command.action(logger, { options: { debug: true, webUrl: 'https://contoso.sharepoint.com/sites/team-a', name: 'home.aspx', id: 'ede2ee65-157d-4523-b4ed-87b9b64374a6', webPartProperties: '{}' } }, (err?: any) => {
+      try {
+        done();
+      } catch (e) {
+        done(e);
+      }
+    });
+  });
+
+
+
+  it('correctly handles OData error when retrieving pages', (done) => {
+    sinon.stub(request, 'get').callsFake((opts) => {
+      return Promise.reject({ error: { 'odata.error': { message: { value: 'An error has occurred' } } } });
+    });
+
+    command.action(logger, { options: { debug: true, id: '3ede60d3-dc2c-438b-b5bf-cc40bb2351e5', webPartData: "{}", webUrl: 'https://contoso.sharepoint.com', name: 'home.aspx' } } as any, (err?: any) => {
+      try {
+        assert.strictEqual(JSON.stringify(err), JSON.stringify(new CommandError('An error has occurred')));
+        done();
+      } catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  // OPTIONS
+
+  it('supports debug mode', () => {
+    const options = command.options();
+    let containsOption = false;
+    options.forEach(o => {
+      if (o.option === '--debug') {
+        containsOption = true;
+      }
+    });
+    assert(containsOption);
+  });
+
+  // VALIDATION
+
+  it('fails validation if the specified id is not a valid GUID', () => {
+    const actual = command.validate({ options: { id: 'abc', name: 'home.aspx', webUrl: 'https://contoso.sharepoint.com' } });
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if the specified webPartProperties is not a valid JSON string', () => {
+    const actual = command.validate({ options: { webPartProperties: "abc", id: '3ede60d3-dc2c-438b-b5bf-cc40bb2351e5', name: 'home.aspx', webUrl: 'https://contoso.sharepoint.com' } });
+    assert.notStrictEqual(actual, true);
+  });
+  
+  it('fails validation if the specified webPartData is not a valid JSON string', () => {
+    const actual = command.validate({ options: { webPartData: "abc", id: '3ede60d3-dc2c-438b-b5bf-cc40bb2351e5', name: 'home.aspx', webUrl: 'https://contoso.sharepoint.com' } });
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if the webPartData and webPartProperties options are provided', () => {
+    const actual = command.validate({ options: { webPartProperties: "{}", webPartData: "{}", id: '3ede60d3-dc2c-438b-b5bf-cc40bb2351e5', webUrl: 'foo', name: 'home.aspx' } });
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if the webUrl option is not a valid SharePoint site URL', () => {
+    const actual = command.validate({ options: { id: '3ede60d3-dc2c-438b-b5bf-cc40bb2351e5', webUrl: 'foo', name: 'home.aspx' } });
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation when right properties are provided', () => {
```

Comment:
We're missing a test case checking passing validation when webPartProperties have been specified